app/web/src/workers/types/dbinterface.ts

```diff
@@ -285,13 +287,75 @@ export interface ComponentQualificationTotals {
   running: number;
 }
 
-export interface Component {
+export interface EddaComponent {
   id: ComponentId;
   name: string;
   color?: string;
   schemaName: string;
   schemaId: SchemaId;
-  schemaVariantId: SchemaVariantId;
+  schemaVariantId: Reference;
+  schemaVariantName: string;
+  schemaVariantDescription?: string;
+  schemaVariantDocLink?: string;
+  schemaCategory: string;
+  hasResource: boolean;
+  qualificationTotals: ComponentQualificationTotals;
+  inputCount: number;
+  // this will only be filled in when it is computed
+  outputCount: number;
+  diffCount: number;
+  rootAttributeValueId: AttributeValueId;
+  domainAttributeValueId: AttributeValueId;
+  secretsAttributeValueId: AttributeValueId;
+  siAttributeValueId: AttributeValueId;
+  resourceValueAttributeValueId: AttributeValueId;
+  resourceDiff: {
+    current?: string;
+    diff?: string;
+  };
+}
+
+export interface SchemaVariant {
+  id: string;
+  schemaVariantId: string;
+  schemaName: string;
+  schemaDocLinks?: string;
+  displayName: string | null;
+  category: string;
+  color: string;
+  link: string | null;
+  description: string | null;
+
+  created_at: IsoDateString;
+  updated_at: IsoDateString;
```

Comment:
I think these stay snake case right?

Reply:
Correct, those dont get camelCased by the backend